avocado/core/loader.py

```diff
@@ -235,6 +235,10 @@ def load_test(self, test_factory):
         :return: an instance of :class:`avocado.core.test.Test`.
         """
         test_class, test_parameters = test_factory
+        # discard tags, as they are *not* intented to be parameters
+        # for the test, but used previously during filtering
+        if 'tags' in test_parameters:
+            del(test_parameters['tags'])
```

Comment:
Hmm, this is not really flexible solution. Ideally I'd prefer modifying the whole `loaders` to allow three arguments, where the third argument would be loaders-options/kwargs, but I understand that that might affect other the existing loader plugins (still I believe it'd be the cleanest variant).

Alternatively I'd at least define something like `test_parameters["extra_kwargs"] = {}`, which might optionally include the `tags` and in the future it might also include any other non-test key-word argument.

Once again, the first way would be cleaner, faster and could be implemented with defaults to avoid the immediate need to modify existing loaders.